### PR TITLE
kill monitor process, update debug adapter

### LINKED ISSUE #305: Espressif IDF extention issue flsah / monitor (VSC-558)
I am using Espressif IDF extention v0.6.1.
After flashing, i open the monitor which works just fine.
However, if I recompile and try to flash again I get an error saying "SerialException: could not open port..."
I know this is because it did not close the com port before it started flashing.
Can you guys fix this please, this was an issue before and was fixed but now its back again.

### PR DESCRIPTION
**Pull request by brianignacio5**
Kill ESP-IDF Monitor process when flash is called.

Fix #305 

Update esp-dap-adapter to latest commit.

Can you give it a try @versamodule ?

### CODE REVIEW

**Comment by versamodule** [comment]
Sure, can you tell me how to "Update esp-dap-adapter to latest commit." ?

**Comment by brianignacio5** [comment]
You can try 

```
cd ${CONTAINER_DIR}/vscode-esp-idf-extension/esp-debug-adapter
git pull origin master
```

**Comment by versamodule** [comment]
I get: 
C:\Users\Jim\ESP_4.2\VSCode Portable\data\extensions\espressif.esp-idf-extension-0.6.1>git pull origin master
fatal: not a git repository (or any of the parent directories): .git

**Comment by brianignacio5** [comment]
Oh I see. Well you are using the one already installed by marketplace instead of this PR. Maybe you can try:

1) [Build this PR from source code](https://github.com/espressif/vscode-esp-idf-extension/blob/master/docs/INSTALL.md#build-from-source-code)
2) Install using the extension installer (vsix) generated from this PR [here](https://github.com/espressif/vscode-esp-idf-extension/suites/1998412477/artifacts/39899358) and install using these [steps](https://github.com/espressif/vscode-esp-idf-extension/blob/master/docs/INSTALL.md#install-from-vsix-file)

File: src/extension.ts [hunk 1]
```diff
@@ -342,13 +346,6 @@ export async function activate(context: vscode.ExtensionContext) {
   });
   context.subscriptions.push(sdkDeleteWatchDisposable);
 
-  vscode.window.onDidCloseTerminal((terminal: vscode.Terminal) => {
-    terminal.dispose();
-    setTimeout(() => {
-      monitorTerminal = undefined;
-    }, 200);
-  });
-
```
Comment (pwmb) [inline src/extension.ts:351]: Any reason for this code removal?

Comment (brianignacio5) [reply src/extension.ts:351]: Well I just noticed this code and saw it is redundant. Calling terminal.dispose on OnDidCloseTerminal seems like is not necessary and also monitorTerminal is set as undefined for any terminal being close, which is incorrect.

**Review by pwmb — APPROVED** [review]

**Pull request merged**